src/java/org/apache/cassandra/locator/AbstractCloudMetadataServiceConnector.java

```diff
@@ -25,22 +25,41 @@
 import java.net.URL;
 import java.util.Map;
 
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
 import com.google.common.collect.ImmutableMap;
 
 import static java.nio.charset.StandardCharsets.UTF_8;
 
 abstract class AbstractCloudMetadataServiceConnector
 {
+    static final String METADATA_URL_PROPERTY = "metadata_url";
+
+    public static class DefaultCloudMetadataServiceConnector extends AbstractCloudMetadataServiceConnector
+    {
+        protected DefaultCloudMetadataServiceConnector(String metadataServiceUrl)
+        {
+            super(metadataServiceUrl);
+        }
+    }
+
     protected final String metadataServiceUrl;
 
+    @VisibleForTesting // for mockito
+    protected String getMetadataServiceUrl()
+    {
+        return metadataServiceUrl;
+    }
+
     protected AbstractCloudMetadataServiceConnector(String metadataServiceUrl)
     {
+        Preconditions.checkNotNull(metadataServiceUrl, METADATA_URL_PROPERTY + " for a connector can not be null");
         this.metadataServiceUrl = metadataServiceUrl;
     }
 
     public String apiCall(String query) throws IOException
     {
-        return apiCall(metadataServiceUrl, query, 200);
+        return apiCall(getMetadataServiceUrl(), query, 200);
```

Comment:
why?

Reply:
This is here due to tests. I had to mock what would that method return, this can not be done on a field.

Reply:
Nevertheless, I think this is not needed. I saw your mock and the reason why it fails with NPE is passing `anyString()` as url arg for `apiCall` - simply change it to `any()` and it will accept nulls.